tests/integration/test_fim/test_stats_integrity_sync/test_stats_integrity_sync.py

```diff
@@ -0,0 +1,716 @@
+    attempts_info : shared dict
+        Dictionary with a flag that indicates if the stats collector must start.
+    database_params : dict
+        Database params to be applied for the current test.
+
+    """
+    if case == Cases.case0.value:
+        alerts = open(ALERT_FILE_PATH, 'w')
+        alerts.close()
+
+        # Detect that the agent are been restarted
+        def callback_detect_agent_restart(line):
+            try:
+                return re.match(rf'.*\"agent\":{{\"id\":\"({agent_id})\".*', line).group(1)
+            except (IndexError, AttributeError):
+                pass
+
+        FileMonitor(ALERT_FILE_PATH).start(timeout=max_time_for_agent_setup,
+                                           callback=callback_detect_agent_restart).result()
+
+    modify_database(agent_id, **database_params)
+
+    while True:
+        if n_attempts(agent_id) > 0 and agents_dict[agent_id]['start'] == 0.0:
+            if not attempts_info['start']:
+                with open(filename, 'w') as info_agent:
+                    info_agent.write("agent_id,n_attempts,n_completions,start_time,end_time,total_time,state\n")
+            agents_dict[agent_id]['start'] = time.time()
+            attempts_info['start'] = True
+            print(f'[AGENT] Agent {agent_id} started at {agents_dict[agent_id]["start"]}')
+        actual_n_attempts = n_attempts(agent_id)
+        if (n_completions(agent_id) > 0 or actual_n_attempts > max_n_attempts) and \
+                agents_dict[agent_id]['start'] != 0.0:
+            state = 'complete' if actual_n_attempts < max_n_attempts else 'except_max_attempts'
+            info_collector(agents_dict[agent_id], agent_id, filename, attempts_info, state=state)
+            break
+
+
+def info_collector(agent, agent_id, filename, attempts_info, state='complete'):
+    """Write the stats of the agent during the test.
+
+    This stats will be written when the agent finish its test process (n_completions(agent_id) > 0).
+
+    Parameters
+    ----------
+    agent : dict
+        Individual dictionary with the start time of an agent.
+    agent_id : str
+        Agent id.
+    filename : str
+        Path of the agent's info file.
+    attempts_info : shared dict
+        Dictionary with a flag that indicates if the stats collector must start.
+    state : str
+        complete of except_max_attempts: Indicates if the agent took less attempts than the defined limits or not.
+
+    """
+    if state != 'complete':
+        attempts_info['except_max_attempts'] = True
+        attempts_info['agents_failed'] += 1
+    with open(filename, 'a') as info_agent:
+        print(
+            f"[AGENT] Info {agent_id} writing: "
+            f"{agent_id},{n_attempts(agent_id)},{n_completions(agent_id)},{agent['start']},"
+            f"{time.time()},{time.time() - agent['start']},{state}")
+        info_agent.write(f"{agent_id},{n_attempts(agent_id)},{n_completions(agent_id)},{agent['start']},"
```

Comment:
Store the stats in a proper data structure (i.e. pandas DataFrame) and then dump to a CSV file.